elasticsearch-aws/src/main/scala/com/sumologic/elasticsearch/util/AwsRequestSigner.scala

```diff
@@ -25,19 +25,33 @@ import java.util.{Calendar, TimeZone}
 import javax.crypto.Mac
 import javax.crypto.spec.SecretKeySpec
 
-import com.amazonaws.auth.{AWSCredentials, AWSSessionCredentials}
+import com.amazonaws.auth.{AWSCredentials, AWSCredentialsProvider, AWSSessionCredentials}
+import com.amazonaws.internal.StaticCredentialsProvider
 import com.sumologic.elasticsearch.restlastic.RequestSigner
 import spray.http.HttpHeaders.{Host, RawHeader}
 import spray.http.HttpRequest
 
 /**
  * Sign AWS requests following the instructions at http://docs.aws.amazon.com/general/latest/gr/sigv4_signing.html
- * @param awsCredentials AWS creds to sign with
+ * @param awsCredentialsProvider Interface for providing AWS credentials
  * @param region Region to connect to
  * @param service Service to connect to (eg. "es" for elasticsearch) [http://docs.aws.amazon.com/general/latest/gr/rande.html]
  */
-class AwsRequestSigner(awsCredentials: AWSCredentials, region: String, service: String) extends RequestSigner {
+class AwsRequestSigner(awsCredentialsProvider: AWSCredentialsProvider, region: String, service: String) extends RequestSigner {
   val Algorithm = "AWS4-HMAC-SHA256"
+  require(Option(awsCredentialsProvider.getCredentials).isDefined, "awsCredentialsProvider must return non null awsCredentials.")
```

Comment:
Although I can see why you did this, seems like `!= Null`  would be a bit clearer